cumulus_library_umls/ancilary_tables.sql

```diff
@@ -0,0 +1,349 @@
+
+CREATE TABLE IF NOT EXISTS umls__icd10_block AS
+SELECT DISTINCT
+    r.rui,
+    r.cui1,
+    r.cui2,
+    c.tty,
+    c.code_len,
+    c.code,
+    c.str
+FROM umls__mrrel_icd10cm AS r,
+    umls__mrconso_icd10cm AS c,
+    umls__icd10_chapter AS par
+WHERE
+    c.tty IN ('HT')
+    AND r.rel = 'CHD'
+    AND c.code LIKE '%-%'
+    AND r.cui1 = par.cui2
+    AND r.cui2 = c.cui
+ORDER BY c.code ASC;
+
+CREATE TABLE IF NOT EXISTS umls__icd10_category AS
+SELECT DISTINCT
+    r.rui,
+    r.cui1,
+    r.cui2,
+    c.tty,
+    c.code_len,
+    c.code,
+    c.str
+FROM umls__mrrel_icd10cm AS r,
+    umls__mrconso_icd10cm AS c,
+    umls__icd10_block AS par
+WHERE
+    c.tty IN ('HT', 'PT')
+    AND c.code_len = 3
+    AND r.rel = 'CHD'
+    AND r.cui1 = par.cui2
+    AND r.cui2 = c.cui
+ORDER BY c.code ASC;
+
+CREATE TABLE IF NOT EXISTS umls__icd10_code AS
+
+WITH code_5 AS (
+    SELECT DISTINCT
+        r.rui,
+        r.cui1,
+        r.cui2,
+        c.tty,
+        c.code_len,
+        c.code,
+        c.str
+    FROM umls__mrrel_icd10cm AS r,
+        umls__mrconso_icd10cm AS c,
+        umls__icd10_category AS par
+    WHERE
+        c.tty IN ('HT', 'PT')
+        AND c.code LIKE '%.%'
+        AND (c.code_len = 5 OR c.code LIKE '%.%X%')
+        AND r.rel = 'CHD'
+        AND r.cui1 = par.cui2
+        AND r.cui2 = c.cui
+),
+
+code_6 AS (
+    SELECT DISTINCT
+        r.rui,
+        r.cui1,
+        r.cui2,
+        c.tty,
+        c.code_len,
+        c.code,
+        c.str
+    FROM umls__mrrel_icd10cm AS r,
+        umls__mrconso_icd10cm AS c,
+        code_5 AS par
+    WHERE
+        c.tty IN ('HT', 'PT')
+        AND c.code LIKE '%.%'
+        AND (c.code_len = 6 OR c.code LIKE '%.%X%')
+        AND r.rel = 'CHD'
+        AND r.cui1 = par.cui2
+        AND r.cui2 = c.cui
+),
+
+code_7 AS (
+    SELECT DISTINCT
+        r.rui,
+        r.cui1,
+        r.cui2,
+        c.tty,
+        c.code_len,
+        c.code,
+        c.str
+    FROM umls__mrrel_icd10cm AS r,
+        umls__mrconso_icd10cm AS c,
+        code_6 AS par
+    WHERE
+        c.tty IN ('HT', 'PT')
+        AND c.code LIKE '%.%'
+        AND (c.code_len = 7 OR c.code LIKE '%.%X%')
+        AND c.code LIKE '%.%'
+        AND r.rel = 'CHD'
+        AND r.cui1 = par.cui2
+        AND r.cui2 = c.cui
+    ORDER BY c.code ASC
+),
+
+code_8 AS (
+    SELECT DISTINCT
+        r.rui,
+        r.cui1,
+        r.cui2,
+        c.tty,
+        c.code_len,
+        c.code,
+        c.str
+    FROM umls__mrrel_icd10cm AS r,
+        umls__mrconso_icd10cm AS c,
+        code_7 AS par
+    WHERE
+        c.tty IN ('HT', 'PT')
+        AND c.code LIKE '%.%'
+        AND (c.code_len = 8 OR c.code LIKE '%.%X%')
+        AND c.code LIKE '%.%'
+        AND r.rel = 'CHD'
+        AND r.cui1 = par.cui2
+        AND r.cui2 = c.cui
+    ORDER BY c.code ASC
+)
+
+SELECT
+    rui,
+    cui1,
+    cui2,
+    tty,
+    code_len,
+    code,
+    str,
+    5 AS depth
+FROM code_5
+UNION
```

Comment:
I was recently reminded that Athena [prefers UNION ALL to UNION](https://docs.aws.amazon.com/athena/latest/ug/performance-tuning-query-optimization-techniques.html). You probably don't need de-duplication here, so could use UNION ALL.